src/utils/helpers.js

```diff
@@ -0,0 +1,54 @@
+import { buildDateTimeRangeQuery } from '../components/AllJobLogsView/CustomQueryBuilder';
+import { JOB_EXECUTION_STATUSES, JOB_LOGS_STATUS_QUERY_VALUE } from './constants';
+
+export const getFormattedFilters = (filters) => Object.entries(filters).reduce((acc, [key, value]) => {
+  if (Array.isArray(value)) {
+    if (value.length === 1) {
+      acc[key] = value[0];
+    } else {
+      acc[key] = value;
+    }
+  } else {
+    acc[key] = value;
+  }
+
+  return acc;
+}, {});
+
+export const getMappedFilters = (timezone) => ({
+  completedDate: buildDateTimeRangeQuery.bind(null, ['completedDate'], timezone),
+  startedDate: buildDateTimeRangeQuery.bind(null, ['startedDate'], timezone),
+  status: query => {
+    switch (true) {
+      case query === JOB_EXECUTION_STATUSES.FAIL:
+        return 'status=(FAIL or COMPLETED_WITH_ERRORS)';
+      case Array.isArray(query):
+        return `status=(${query.map(v => (
+          v === JOB_EXECUTION_STATUSES.COMPLETED
+            ? JOB_LOGS_STATUS_QUERY_VALUE
+            : `"${v}"`)).join(' or ')})`;
+      default:
+        return `status=${query}`;
+    }
+  },
+  hrId: query => `hrid="${query}"`,
+  'runBy.userId': query => `runById="${query}"`,
+});
+
+export const getSortedUsers = (users = []) => users.map(item => {
+  return {
+    userId: item.userId,
+    firstName: item.firstName,
+    lastName: item.lastName,
+  };
+})
+  .sort((userA, userB) => {
+    const nameA = userA.firstName || userA.lastName;
+    const nameB = userB.firstName || userB.lastName;
+
+    if (userA.firstName?.localeCompare(userB.firstName) === 0) {
+      return userA.lastName.localeCompare(userB.lastName);
+    }
+
+    return nameA.localeCompare(nameB);
+  });
```

Comment:
Sorting by name is not a very "data-export-y" function, and it's surprising to me that we would want to sort by first-name rather than last-name. There's nothing wrong with this, obviously, it works as it should. So my question is not "does this work correctly?" but "is it the right thing to do?" (maybe, but it would surprise me) and also "is this the right place to do it?" (probably not, I think a better place for this would be [stripes-util where we have related functions like `getFullName()`](https://github.com/folio-org/stripes-util/blob/2be3794d455161c34fc1a6f957c7555d4b03fa30/lib/getFullName.js).

Reply:
To be honest, I just moved this function to helpers and did not go into detail about why this is so. I made a note, I'll check with our PO and we'll fix it in future stories. Thanks!